frontend/src/components/ui/ScorePhaseModal.jsx

```diff
@@ -1,23 +1,95 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { BUILD } from "../../Game";
+import { TOWERS } from "../../entities/buildables";
+import Modal from 'react-bootstrap/Modal';
 
-export const ScorePhaseModal = ({ game }) => {
+export const ScorePhaseModal = ({ game, show }) => {
     const [copied, setCopied] = useState(false);
+    const [rewardsChosen, setRewardsChosen] = useState([]);
     const handleClick = () => game.setPhase(BUILD);
 
+    const CASTLE_HEAL = 200;
+    const REWARD_LIMIT = 1; // number of rewards to choose
+    const BLUEPRINT_REWARDS = 2; // number of blueprints in rewards
+    const towerKeys = Array.from(TOWERS.keys());
+    const BASE_REWARDS = [
+        {
+            label: `+${game.goldReward} Gold`,
+            fn: () => game.gold += game.goldReward
+        },
+        {
+            label: `+${CASTLE_HEAL} Castle HP`,
+            fn: () => {
+                game.castle.hp += CASTLE_HEAL;
+                game.castle.hp = Math.min(game.castle.maxHp, game.castle.hp);
+            }
+        },
+        // {
+        //     label: "+2 Tower Limit",
+        //     fn: () => game.towerLimit += 2
+        // },
+    ];
+    const [rewards, setRewards] = useState(BASE_REWARDS);
+
+    useEffect(() => {
+        const newRewards = [...BASE_REWARDS];
+        for (let i = 0; i < BLUEPRINT_REWARDS; i++) {
+            if (game.blueprints.size < towerKeys.length) {
+                let blueprintIndex = 0;
+                while (game.blueprints.has(towerKeys[blueprintIndex] || newRewards.at(-1).label === towerKeys[blueprintIndex])) {
+                    blueprintIndex = Math.floor(Math.random() * towerKeys.length);
+                }
+                // newRewards.push({
+                //     label: `Blueprint: ${towerKeys[blueprintIndex]}`,
+                //     fn: () => game.blueprints.add(towerKeys[blueprintIndex])
+                // });
```

Comment:
Instead of commenting this out, prolly better to set `BLUEPRINT_REWARDS = 0` since the surrounding code is also blueprint related